src/data/pi.lean

```diff
@@ -111,6 +111,30 @@ variables (f)
 lemma single_injective (i : I) : function.injective (single i : f i → Π i, f i) :=
 function.update_injective _ i
 
+section extend
+
+@[to_additive]
+lemma _root_.function.extend_one {α β γ} [has_one γ] (f : α → β) :
+  function.extend f (1 : α → γ) (1 : β → γ) = 1 :=
+funext $ λ _, by apply if_t_t _ _
+
+@[to_additive]
+lemma _root_.function.extend_mul {α β γ} [has_mul γ] (f : α → β) (g₁ g₂ : α → γ) (e₁ e₂ : β → γ) :
+  function.extend f (g₁ * g₂) (e₁ * e₂) = function.extend f g₁ e₁ * function.extend f g₂ e₂ :=
+funext $ λ _, by convert (apply_dite2 (*) _ _ _ _ _).symm
+
+@[to_additive]
+lemma _root_.function.extend_inv {α β γ} [has_inv γ] (f : α → β) (g : α → γ) (e : β → γ) :
+  function.extend f (g⁻¹) (e⁻¹) = (function.extend f g e)⁻¹ :=
+funext $ λ _, by convert (apply_dite has_inv.inv _ _ _).symm
+
+@[to_additive]
+lemma _root_.function.extend_div {α β γ} [has_div γ] (f : α → β) (g₁ g₂ : α → γ) (e₁ e₂ : β → γ) :
+  function.extend f (g₁ / g₂) (e₁ / e₂) = function.extend f g₁ e₁ / function.extend f g₂ e₂ :=
+funext $ λ _, by convert (apply_dite2 (/) _ _ _ _ _).symm
+
+end extend
```

Comment:
It probably makes sense to move this outside the `namespace pi` and add a `namespace function` with `variables {α β γ : Type*}`